src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/cos/CertificateOfServiceGenerator.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.hmcts.reform.unspec.service.docmosis.cos;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.unspec.enums.ServedDocuments;
+import uk.gov.hmcts.reform.unspec.enums.ServiceLocationType;
+import uk.gov.hmcts.reform.unspec.model.Address;
+import uk.gov.hmcts.reform.unspec.model.CaseData;
+import uk.gov.hmcts.reform.unspec.model.ServiceLocation;
+import uk.gov.hmcts.reform.unspec.model.SolicitorReferences;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisDocument;
+import uk.gov.hmcts.reform.unspec.model.docmosis.cos.CertificateOfServiceForm;
+import uk.gov.hmcts.reform.unspec.model.docmosis.sealedclaim.Representative;
+import uk.gov.hmcts.reform.unspec.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.unspec.model.documents.DocumentType;
+import uk.gov.hmcts.reform.unspec.model.documents.PDF;
+import uk.gov.hmcts.reform.unspec.service.docmosis.DocumentGeneratorService;
+import uk.gov.hmcts.reform.unspec.service.docmosis.TemplateDataGenerator;
+import uk.gov.hmcts.reform.unspec.service.documentmanagement.DocumentManagementService;
+import uk.gov.hmcts.reform.unspec.utils.CaseNameUtils;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+import static uk.gov.hmcts.reform.unspec.enums.ServedDocuments.OTHER;
+import static uk.gov.hmcts.reform.unspec.service.docmosis.DocmosisTemplates.N215;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CertificateOfServiceGenerator extends TemplateDataGenerator<CertificateOfServiceForm> {
+
+    private static final Representative TEMP_REPRESENTATIVE = Representative.builder()
+        .contactName("MiguelSpooner")
+        .dxAddress("DX 751Newport")
+        .organisationName("DBE Law")
+        .phoneNumber("0800 206 1592")
+        .emailAddress("jim.smith@slatergordon.com")
+        .serviceAddress(Address.builder()
+                            .addressLine1("AdmiralHouse")
+                            .addressLine2("Queensway")
+                            .postTown("Newport")
+                            .postCode("NP204AG")
+                            .build())
+        .build(); //TODO Rep details need to be picked from reference data
+    public static final String TEMP_REFERENCE_NUMBER = "000LR001"; //TODO Need to agree a way to get
+    public static final LocalDate TEMP_SUBMITTED_ON = LocalDate.now(); //TODO: Not populated yet in case data
```

Comment:
claim issue date? This should be stored now

Reply:
In existing legal service, we have two dates, one for `submitted on` and other  for `issued at`. Both can be different. i.e. if a user submits the claim on Friday after 16:00, the submitted date would be Friday and issued on would be Monday. Unless those semantice are changed.